meetings/2021-07/july-15.md

```diff
@@ -0,0 +1,734 @@
+
+WH: I'm also a bit unclear as to what is going into stage one and I would note that there is a deadline to post documents a number of days in advance of a meeting so we can review them.
+
+JHD:  Sure. And I mean that and the lack of materials are both valid, Reasons to procedurally, reject stage 1. I just but that doesn't mean you have to reject it but it sounds it sounds like they're not, there's at least a few people that aren't comfortable with stage one.
+
+WH: Yeah, but I also want to mention that I do like where you're going with this. 
+
+JWK: So I didn't catch up with the conversation before. What is the use case? Why do we need an API for it? 
+
+JHD: So that people can write code, that can't be trivially broken by someone later messing with the environment like node core uses this, because if do delete function.prototype.call node core just craps itself and dies immediately. And so there's all over nerdcore. They're trying to slowly patch this problem so that its robust against user modification of the platform and all of my npm packages are written in ways where as long as they run first. In a before user code is mess with anything, then they are robust against user code messing with things. And get intrinsic would be a much more convenient and potentially more performant although that's less of a concern for me way to to write these this robust code.
+
+ JWK:I guess that must be able to mock otherwise it might break visualization.
+
+KG: Right. So the assumption is that you are running first so that you can save off a copy of getOriginals that no one else will thereafter be in a position to modify.
+
+JWK: if they are running with why need you can just grab those by yourself
+
+JHD: I might Define a function first by the time that functions run a bunch of add code might have run or user extensions in their browser and mess with stuff and the in order for the the body of the function to be robust, I have to have cashed everything that it's going to use, and in to protect methods, I have to call bound to them in a way that they don't rely on.collared. Fine. Being present at runtime. And there's, I can already do this. I have a get intrinsic library that does this, but having it built into the platform, just like the Realms, proposal would make things a lot simpler and harder to get wrong.
+
+JWK: The realms use case seems much larger than getIntrinsics. It seems like a very limited use case. I don't know if it's worth it to do that.
+
+JHD: Yes, it's a fair point that there is a small number of people that may utilize that API, but I mean the that is a significant percentage of npm, download traffic already uses my get intrinsic Library so like it's the end users will will benefit from that. A lot of end users will benefit from the code even if they don't use it directly while. I think this may be true of realms.
```

Comment:
```suggestion
JHD: Yes, it's a fair point that there is a small number of people that may utilize that API, but I mean the that is a significant percentage of npm's download traffic already uses my `get-intrinsic` library so the end users will benefit from that. A lot of end users will benefit from the code even if they don't use it directly. I think this may also be true of Realms.
```